pkg/kubelet/network/dns/dns.go

```diff
@@ -99,19 +102,31 @@ func omitDuplicates(strs []string) []string {
 func (c *Configurer) formDNSSearchFitsLimits(composedSearch []string, pod *v1.Pod) []string {
 	limitsExceeded := false
 
-	if len(composedSearch) > validation.MaxDNSSearchPaths {
-		composedSearch = composedSearch[:validation.MaxDNSSearchPaths]
+	for i, search := range composedSearch {
+		if len(search) > utilvalidation.DNS1123SubdomainMaxLength {
+			composedSearch = composedSearch[:i]
```

Comment:
In this case I think it's safer to leave it in, invalid or not.  Truncating is pretty hostile.

Reply:
is not better to just skip only the ones that exceeds the limits and don't break?
if I have a search with"
s1 badSearch300chars s2 s3 s4

I will have s1 s2 s3 s4, with this I'll have s1 IIUIRC

Reply:
https://github.com/golang/go/wiki/SliceTricks#filter-in-place

Reply:
Done.